File: iterator/python/implementations/physical_book_collection.py
from typing import List
from interfaces.book_collection import BookCollection
from interfaces.iterator import Iterator
from interfaces.book import Book
from implementations.physical_book_iterator import PhysicalBookIterator

class PhysicalBookCollection(BookCollection):
    def __init__(self):
        self._books: List[Book] = []

    def add_book(self, book: Book) -> None:
        self._books.append(book)

    def create_iterator(self) -> Iterator:
        return PhysicalBookIterator(self._books)

File: iterator/java/implementations/PhysicalBookCollection.java
package iterator.java.implementations;

import java.util.ArrayList;
import java.util.List;

import iterator.java.interfaces.BookCollection;
import iterator.java.interfaces.Iterator;
import iterator.java.interfaces.Book;

public class PhysicalBookCollection implements BookCollection {
    private List<Book> books = new ArrayList<>();

    public void addBook(Book book) {
        books.add(book);
    }

    @Override
    public Iterator createIterator() {
        return new PhysicalBookIterator(books);
    }
}




